### PR TITLE
feat: add logs for scripts

### PR DESCRIPTION
**Pull request by hal3e**
closes: https://github.com/FuelLabs/fuels-rs/issues/713

- Script calls can use the same logging interface as contract calls
- Added log tests for scripts
- Added require log tests for scripts
- Update documentation
- Renamed the `logging` and `logged_types` sway projects for clarity
- Moved `decode_revert_error()` to `logs.rs` -> open for suggestions where to put it...

### CODE REVIEW

File: docs/src/scripts/scripts.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# Running scripts
```
Comment (iqdecay) [inline docs/src/scripts/scripts.md:1]: Obsolete file

File: packages/fuels-contract/src/contract.rs [hunk 2]
```diff
@@ -481,18 +483,6 @@ pub fn get_decoded_output(
     Ok(decoded_value)
 }
 
-// Decode the logged type from the receipt of a `RevertTransactionError` if available
-fn decode_revert_error(err: Error, log_decoder: &LogDecoder) -> Error {
```
Comment (iqdecay) [inline packages/fuels-contract/src/contract.rs:485]: Moving this function makes complete sense.

File: packages/fuels-contract/src/logs.rs [hunk 3]
```diff
@@ -85,3 +85,15 @@ impl LogDecoder {
             .extend(log_decoder.logs_map.clone().into_iter());
     }
 }
+
+// Decode the logged type from the receipt of a `RevertTransactionError` if available
```
Comment (iqdecay) [inline packages/fuels-contract/src/logs.rs:89]:
```suggestion
/// Decodes the logged type from the receipt of a `RevertTransactionError` if available
```

File: packages/fuels-core/src/code_gen/abigen.rs [hunk 4]
```diff
@@ -172,11 +174,17 @@ impl Abigen {
...
                 }
 
                 impl #name {
                     pub fn new(wallet: WalletUnlocked, binary_filepath: &str) -> Self {
-                        Self {wallet: wallet, binary_filepath: binary_filepath.to_string()}
+                        Self {
+                            wallet: wallet,
+                            binary_filepath: binary_filepath.to_string(),
+                            logs_map: get_logs_hashmap(&[#(#log_id_param_type_pairs),*],
+                                                       &Bech32ContractId::from(ContractId::zeroed()))
```
Comment (iqdecay) [inline packages/fuels-core/src/code_gen/abigen.rs:186]: This makes me think we should accept an `Option<Bech32ContractID>` for the `get_logs_hashmap` function: it makes no sense to pass in a null contract_id, it should be hidden away in the function itself. Wdyt?
Also this way we don't have to have the `Bech32ContractID` include statement in the general case, which makes more sense.

File: packages/fuels/tests/logs.rs [hunk 5]
```diff
@@ -439,5 +441,187 @@ async fn test_multi_call_require_log_multi_contract() -> Result<(), Error> {
...
+    );
+
+    let wallet = launch_provider_and_get_wallet().await;
+    let bin_path = "../fuels/tests/logs/script_logs/out/debug/script_logs.bin";
+    let instance = log_script::new(wallet.clone(), bin_path);
+
+    let response = instance.main().call().await?;
+
+    let logs = response.get_logs()?;
+    let _log_u64 = response.get_logs_with_type::<u64>()?;
```
Comment (iqdecay) [inline packages/fuels/tests/logs.rs:463]:
```suggestion
    let log_u64 = response.get_logs_with_type::<u64>()?;
```

File: packages/fuels/tests/logs.rs [hunk 6]
```diff
@@ -439,5 +441,187 @@ async fn test_multi_call_require_log_multi_contract() -> Result<(), Error> {
 
         assert_is_revert_containing_msg("StructDeeplyNestedGeneric", error);
     }
+
+    Ok(())
+}
+
+#[tokio::test]
```
Comment (iqdecay) [inline packages/fuels/tests/logs.rs:448]:
```suggestion
#[tokio::test]
#![allow(unused_variables)]
```

File: packages/fuels/tests/logs.rs [hunk 7]
```diff
@@ -439,5 +441,187 @@ async fn test_multi_call_require_log_multi_contract() -> Result<(), Error> {
...
+        let error = instance
+            .main(MatchEnum::RequireCustomGeneric())
+            .call()
+            .await
+            .expect_err("Should return a revert error");
+
+        assert_is_revert_containing_msg("StructDeeplyNestedGeneric", error);
+    }
+
     Ok(())
```
Comment (iqdecay) [inline packages/fuels/tests/logs.rs:626]: Those tests are really good

**Review by iqdecay — CHANGES_REQUESTED** [review]
Very nice, super clean! Left some comments

File: packages/fuels/tests/logs.rs (hunk 5 above)
Comment (hal3e) [reply packages/fuels/tests/logs.rs:463]: I am not using this field in this test.. but I left it there for the book example. This way I can show both and do not have to make a new anchor.. If we do not like this I will add the second anchor ...

File: packages/fuels/tests/logs.rs (hunk 7 above)
Comment (hal3e) [reply packages/fuels/tests/logs.rs:626]: I have to give credit to @MujkicA. Those are his test but modified for scripts :smile:

**Review by iqdecay — APPROVED** [review]
Nice! LGTM :)

File: packages/fuels-contract/src/script_calls.rs [hunk 8]
```diff
@@ -131,7 +133,9 @@ where
 
     /// Call a script on the node, in a state-modifying manner.
     pub async fn call(self) -> Result<FuelCallResponse<D>, Error> {
-        Self::call_or_simulate(&self, false).await
+        Self::call_or_simulate(&self, false)
+            .await
+            .map_err(|err| decode_revert_error(err, &self.log_decoder))
```
Comment (digorithm) [inline packages/fuels-contract/src/script_calls.rs:138]: I love the reusability here!

**Review by digorithm — APPROVED** [review]
NIce! The tests/docs are awesome.

**Pull request merged**